File: External/meegpipe/+mperl/+file/+spec/catpath.m
function path = catpath(volume, dir, file)
% CATPATH - Volume, directory, and file conversion to full path
%
%
% path = filespec.catpath(volume, dir, file)
%
% Where
%
% VOLUME is the volume name. It is not significant under Unix.
%
% DIR is the directory name.
%
% FILE is the file name
%
%
%
% See also: filespec.splitpath, filespec

% (c) German Gomez-Herrero, german.gomezherrero@kasku.org

% Documentation: pkg_filespec.txt
% Description: Takes volume, directory and file portions and returns path

if nargin < 3 || isempty(file), file = ' '; end
if nargin < 2 || isempty(dir), dir = ' '; end
if nargin < 1 || isempty(volume), volume = ' '; end

path = perl('+mperl/+file/+spec/catpath.pl', volume, dir, file);


end

File: External/meegpipe/+mperl/+file/+spec/catpath.pl
#!/usr/bin/perl
# (c) German Gomez-Herrero, german.gomezherrero@kasku.org

# Description: Takes volume, directory and file portions and returns a path
# Documentation: pkg_filespec.txt

use warnings;

# Make our Spec.pm visible
my $libLocation;
BEGIN {
    use Cwd 'abs_path';
    use File::Spec;
    my $name = abs_path($0);
    $name=~m/^.+\//; 
    $libLocation = abs_path(File::Spec->catdir($&, '../../lib'));
}

use lib $libLocation;
use File::Spec;

my ($volume, $dir, $file) = (shift, shift, shift);

my $path = File::Spec->catpath($volume, $dir, $file);

print $path;
